src/platform/forms/containers/review/SubmitController.jsx

```diff
@@ -0,0 +1,232 @@
+    }
+
+    // Validation errors in this situation are not visible, so we’d
+    // like to know if they’re common
+    const { isValid, errors } = isValidForm(form, pageList);
+    if (!isValid) {
+      recordEvent({
+        event: `${trackingPrefix}-validation-failed`,
+      });
+      Sentry.withScope(scope => {
+        scope.setExtra('errors', errors);
+        scope.setExtra('prefix', trackingPrefix);
+        Sentry.captureMessage('Validation issue not displayed');
+      });
+      props.setSubmission('status', 'validationError');
+      props.setSubmission('hasAttemptedSubmit', true);
+      return;
+    }
+
+    // User accepted if required, and no errors, so submit
+    props.submitForm(formConfig, form);
+  }
+
+  const [activeComponent, setActiveComponent] = useState(
+    <SubmitButtons
+      formConfig={formConfig}
+      goBack={goBack}
+      onSubmit={onSubmit}
+    />,
+  );
+
+  useEffect(
+    () => {
+      switch (status) {
+        case false:
+          setActiveComponent(
+            <SubmitButtons
+              formConfig={formConfig}
+              goBack={goBack}
+              onSubmit={onSubmit}
+            />,
+          );
+          break;
+        case SUBMISSION_STATUSES.applicationSubmitted:
+          {
+            const newRoute = `${formConfig.urlPrefix}confirmation`;
+            setActiveComponent(
+              <ApplicationSubmitted
+                formConfig={formConfig}
+                formSubmission={submission}
+                goBack={goBack}
+                onSubmit={onSubmit}
+              />,
+            );
+            if (status !== prevStatus) router.push(newRoute);
+          }
+          break;
+        case SUBMISSION_STATUSES.clienError:
+          setActiveComponent(
+            <ClientError
+              formConfig={formConfig}
+              goBack={goBack}
+              onSubmit={onSubmit}
+            />,
+          );
+          break;
+        case SUBMISSION_STATUSES.submitPending:
+          setActiveComponent(
+            <SubmitPending
+              formConfig={formConfig}
+              goBack={goBack}
+              onSubmit={onSubmit}
+            />,
+          );
+          break;
+        case SUBMISSION_STATUSES.throttledError:
+          setActiveComponent(
+            <ThrottledError
+              formConfig={formConfig}
+              formSubmission={submission}
+              goBack={goBack}
+              onSubmit={onSubmit}
+            />,
+          );
+          break;
+        case SUBMISSION_STATUSES.validationError:
+          setActiveComponent(
+            <ValidationError
+              formConfig={formConfig}
+              goBack={goBack}
+              onSubmit={onSubmit}
+            />,
+          );
+          break;
+        default:
+          setActiveComponent(<GenericError formConfig={formConfig} />);
+          break;
+      }
+    },
+    [status, prevStatus, form, pageList],
+  );
+
+  return activeComponent;
```

Comment:
useEffect will take a second param with options of the props/things to evaluate. If changed, it will trigger. Component will always return the activeComponent as set by useState (default) or by the  setState method returned by useState (defined as setActiveComponent here).